Quest-storage layout

OctoOsmo opened the pull request:
#22 layout button which change storage view was added

Also saveOrUpdate commonDAO method was fixed by temp workaround

OctoOsmo committed:
change storage layout button with respected methods was added
saveOrUpdate commonDAO method was fixed by temp workaround

OctoOsmo committed:
age limit on grid view was moved into header

Alexander091 commented on dao-manager/dao-manager-api/src/main/java/org/my/adventure/dao_manager/api/dao/CommonDAO.java:
```diff
@@ -9,7 +9,7 @@
 public interface CommonDAO<T extends Common> {
     T getById(Long id);
-    void saveOrUpdate(Common object);
+    Long saveOrUpdate(Common object);
```
Этот хак больше не нужен будет, если ты исправишь всё согласно сообщению из контакта и последующих замечаний

Alexander091 commented on dao-manager/dao-manager-ejb/src/main/java/org/my/adventure/dao_manager/ejb/dao/CommonDAOImpl.java:
```diff
@@ -30,11 +29,17 @@ public T getById(Long id){
         Session session = sessionFactory.openSession();
+        session.beginTransaction();
         session.saveOrUpdate(object);
         session.flush();
```
flush можно убрать. Для дефолтного состояния сброс всех изменений перед сохранением итак предполагается
http://stackoverflow.com/questions/3220336/whats-the-use-of-session-flush-in-hibernate

Alexander091 commented on dao-manager/dao-manager-ejb/src/main/java/org/my/adventure/dao_manager/ejb/dao/CommonDAOImpl.java:
```diff
@@ -30,11 +29,17 @@ public T getById(Long id){
         session.flush();
+        session.getTransaction().commit();
+        object = session.get(type, object.getId());
+        id = object.getId();
```
Это вообще, ужас. Даже с хаком, можно было вернуть id у object. Но загружать только что сохранёный объект, это уже через чур

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/QuestStorageController.java:
```diff
@@ -21,21 +23,63 @@
+    ExternalContext externalContext = null;//TODO: fix layout bug
+    Map<String, Object> sessionMap = null;
+
+    public QuestStorageController() {
```
В этом классе ты смешал всё что только возможно, работу с DAO, подготовка объектов к отображению, какие-то специфичные обработки. В итоге этот класс нарушает один из принципов SOLID, именно Single responsibility principle 
https://ru.wikipedia.org/wiki/SOLID_(%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BD%D0%BE-%D0%BE%D1%80%D0%B8%D0%B5%D0%BD%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%BD%D0%BE%D0%B5_%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5)

В общем нужно разделить этот класс. Собственно из-за того, что сочетает в себе всё, у тебя не получится внедрить локальные бины.
Один класс, должен отвечать за отрисовку, он пойдёт в quest-storage-web. это будет jsf контроллер.
Другой класс будет выполнять бизнес-логику и в нём будет связь с DAO. И вызывать он будет третий класс, если нужно, который преобразует данные согласно твоим нуждам.

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/QuestStorageController.java:
```diff
@@ -21,21 +23,63 @@
+                default: layout = "scroller";
+                    break;
+            }
+        }catch (NullPointerException e){//TODO: add logger
```
никогда и ещё раз никогда так не обрабатывай NPE. В try-catch должны обрабатываться только check исключения.
Чем тебе не подошла обычная проверка на null?

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/QuestStorageController.java:
```diff
@@ -21,21 +23,63 @@
     public List<Quest> getQuests(){
         List<Quest> quests = questDAO.getAll();
+        for (Quest quest : quests) {
+            if (quest.getImage() == null)
```
На мой взгляд не тебе нужно проставлять картинки по умолчанию. При сохранение квеста это должно делаться.
Ну либо эта дефолтная картинка у нас приходит с установкой и тогда хранить в базе её не нужно. А в xhtml проверяется есть ли у квеста image, нет указать путь к дефолтной

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/QuestStorageController.java:
```diff
@@ -21,21 +23,63 @@
+        }
         quests.sort(new Comparator<Quest>() {
             public int compare(Quest o1, Quest o2) {
                 return o1.getId() > o2.getId() ? 1 : (o1.getId() < o2.getId()) ? -1 : 0;
```
Что-то мне подсказывает, что записи и так приходят отсортированные по ID. И вообще это сомнительное занятие специально сортировать по id. Пока нет отдельного функционала сортировки, хотя бы пускай по имени сортируются записи

OctoOsmo committed:
user table was renamed into users, mapping was changed respectfully
 login pages was added
 jaas basic support was added

OctoOsmo committed:
commonDAO was reverted
TODO was added into storage controller

OctoOsmo committed:
project structure was changed
quest-storage-web module was added
QuestStorageController was replaced by QuestController
QuestBean was added

OctoOsmo committed:
database password was changed to my local value

OctoOsmo committed:
Merge branches 'dev-branch' and 'quest-storage' of https://github.com/Alexander091/text-adventure into quest-storage

OctoOsmo committed:
authorization form and 404 page was added

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/org/my/adventure/queststorage/impl/QuestBean.java:
```diff
@@ -0,0 +1,41 @@
+    public void deleteQuest(Long id){
+        System.out.println("deleting " + id+ " " + questDAO.getById(id));
+        try {
+            questDAO.delete(questDAO.getById(id));
```
Используй HQL для удаления, не стоит загружать сущность, чтобы удалить. Каскадное удаление реализовано через базу, верно? не через Hibirnate?

Alexander091 commented on quest-storage/quest-storage-impl/src/main/java/org/my/adventure/queststorage/impl/QuestBean.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    public void deleteQuest(Long id){
+        System.out.println("deleting " + id+ " " + questDAO.getById(id));
```
Используй логер log4j

Alexander091 commented:
Я смёрджу, но понадеюсь, что ты в будущем исправишь замечания
